### DISCUSSION TITLE
Access ENV variables injected by Apache mod_ssl

### DISCUSSION THREAD

**Discussion by Markus-** [PostgREST/postgrest#2074] [Q&A] [answered]
Hi all,

I need to access some of the environment variables injected by an apache webserver in front of PostgREST.

https://httpd.apache.org/docs/2.4/mod/mod_ssl.html

Can anybody give me a hint how to do it? 

I want to use client certificate authentication by apache and use the SSL_CLIENT_* variables.

Regards
Markus

**Comment by steve-chavez** [comment]
Hey @Markus-,

Hm, what's your end goal here? You want to setup HTTPS for PostgREST by using Apache?

(We have a section for nginx: https://postgrest.org/en/v9.0/admin.html#https)

**Comment by Markus-** [comment]
Hi,

as I said, I want to utilize the environment variables Apache adds to the request when authenticating with client certificates.

In PHP I can access those by using the $_SERVER and Flask/Python in request variable.

I would assume in PostgREST there is also a request variable which is probably populated by a proxy?

Regard
Markus

Am 09.12.2021 um 17:56 schrieb Steve Chavez ***@***.***>:

﻿

Hey @Markus-<https://github.com/Markus->,

Hm, what's your end goal here? You want to setup HTTPS for PostgREST by using Apache?

(We have a section for nginx: https://postgrest.org/en/v9.0/admin.html#https)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/PostgREST/postgrest/discussions/2074#discussioncomment-1780546>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACI4VUBJYFTR5CZSSTAN67LUQDNUFANCNFSM5JWNEM3Q>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by laurenceisla** [comment, marked as answer]
I'm not completely sure but it may work using custom environment variables in the configuration settings https://postgrest.org/en/v9.0/configuration.html#app-settings. You'll need to set the variables there before starting PostgREST, e.g. `app.settings.ssl_cert_x=$(SSL_CERT_X)`.

If the env variables can't be accessed system wide, then another option would be to somehow set those variables in a header and then access them using GUC names: https://postgrest.org/en/latest/api.html?highlight=guc#accessing-request-headers-cookies-and-jwt-claims. I don't know how feasible that option is, though.

**Reply by Markus-** [reply]
Hi,

I think your second tip does the trick. If I can access the headers and using a json-cast to access all available fields it should work. I am quite short on time right now, so I will mark this question as answered, because I think this will definitely will do the trick.

Thanks
Markus
